app/grandchallenge/reader_studies/views.py

```diff
@@ -1456,17 +1432,18 @@ def form_valid(self, form):
 
     def get_success_url(self):
         return reverse(
-            "reader-studies:display-set-detail",
-            kwargs={"pk": self.kwargs["pk"], "slug": self.kwargs["slug"]},
+            "reader-studies:display_sets",
+            kwargs={"slug": self.kwargs["slug"]},
         )
 
 
 class DisplaySetInterfacesCreate(ObjectPermissionRequiredMixin, FormView):
```

Comment:
For this view, we're no longer using the post method. We're only using this to get the correct form fields for newly added interfaces. Validating form data and adding the interface and value to the display set is done through the create and update views (they are now fields on the create / update form). So that means that the `form_valid` method here can go, and maybe we should explicitly deny the post method on this view? Something like 

```python
def post(self):
    return HttpResponseNotAllowed()
```

or we raise a RunTimeError so that we are made aware of this happening?

Reply:
The same goes for the `DisplaySetFilesUpdate` view, which I haven't updated yet in the interest of keeping the changeset smaller, but it'll follow the same logic as the `InterfacesCreate` view.

Reply:
Sound like a strange construction but from what you say my guess is that it should be a `TemplateView` where you add in the form handling yourself.

Reply:
Yeah, I guess that makes sense. I will change it to a TemplateView then